cache/transform.go

```diff
@@ -7,6 +7,19 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+func TransformConfigMap(obj any) (any, error) {
+	cm, ok := obj.(*corev1.ConfigMap)
+	if !ok {
+		return nil, fmt.Errorf("%T is not of type 'ConfigMap'", obj)
+	}
+
+	partial := &metav1.PartialObjectMetadata{}
+	partial.SetGroupVersionKind(cm.GroupVersionKind())
+	partial.ObjectMeta = cm.ObjectMeta
+
+	return partial, nil
+}
```

Comment:
What happens if you do call `kubeCache.Get(configMap)`? You should get back a `ConfigMap` with just the metadata set. 

My other thought of stripping the fields on transform like this is that this will only reduce what gets saved in the local kube cache but the full configmap objects will still get sent over the wire. There should be a way to setup the informers such that they only request PartialObjectMetadata but I haven't looked into how to do this either for controller-runtime or client-go.

For now this is fine. You can either keep this as it is or go back to disabling the cache altogether for configmaps which is the change you had originally but keeping using the `in.kialiSAClients` to fetch the configmap in discovery.

Reply:
With current `partial := &metav1.PartialObjectMetadata{}` the `kubeCache.Get(configMap)` comes with a warning `Unable to set control plane configuration: cache had type *v1.PartialObjectMetadata, but *v1.ConfigMap was asked for"}`.
And yes, even with a fixed TransformConfigMap all the data is transfered by the wire then filtered out while setting in a cache. And then when necessary is called directly from the API. Which does not make sense.

So I switched to the solution of Disabling cache for ConfigMaps but keeping direct calls to API via `in.kialiSAClients`.